app/src/main/java/io/github/droidkaigi/confsched2017/view/motion/GravityArcMotion.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.droidkaigi.confsched2017.view.motion;
+
+import android.content.Context;
+import android.graphics.Path;
+import android.transition.ArcMotion;
+import android.util.AttributeSet;
+
+/**
+ * A tweak to {@link ArcMotion} which slightly alters the path calculation. In the real world
+ * gravity slows upward motion and accelerates downward motion. This class emulates this behavior
+ * to make motion paths appear more natural.
+ * <p>
+ * See https://www.google.com/design/spec/motion/movement.html#movement-movement-within-screen-bounds
+ */
+public class GravityArcMotion extends ArcMotion {
+
+    private static final float DEFAULT_MIN_ANGLE_DEGREES = 0;
+    private static final float DEFAULT_MAX_ANGLE_DEGREES = 70;
+    private static final float DEFAULT_MAX_TANGENT = (float)
+            Math.tan(Math.toRadians(DEFAULT_MAX_ANGLE_DEGREES / 2));
+
+    private float mMinimumHorizontalAngle = 0;
+    private float mMinimumVerticalAngle = 0;
+    private float mMaximumAngle = DEFAULT_MAX_ANGLE_DEGREES;
+    private float mMinimumHorizontalTangent = 0;
+    private float mMinimumVerticalTangent = 0;
+    private float mMaximumTangent = DEFAULT_MAX_TANGENT;
+
+    public GravityArcMotion() {
+    }
+
+    public GravityArcMotion(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public void setMinimumHorizontalAngle(float angleInDegrees) {
+        mMinimumHorizontalAngle = angleInDegrees;
+        mMinimumHorizontalTangent = toTangent(angleInDegrees);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public float getMinimumHorizontalAngle() {
+        return mMinimumHorizontalAngle;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public void setMinimumVerticalAngle(float angleInDegrees) {
+        mMinimumVerticalAngle = angleInDegrees;
+        mMinimumVerticalTangent = toTangent(angleInDegrees);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public float getMinimumVerticalAngle() {
+        return mMinimumVerticalAngle;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public void setMaximumAngle(float angleInDegrees) {
+        mMaximumAngle = angleInDegrees;
+        mMaximumTangent = toTangent(angleInDegrees);
```

Comment:
**[Android Lint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Error: Call requires API level 21 (current min is 16): android.transition.ArcMotion#toTangent [NewApi]